### DISCUSSION TITLE
Eloquent AsEncrypted*  casts don't use Model::encryptUsing

### DISCUSSION THREAD

**Discussion by krmroland** [laravel/framework#47224] [Ideas]
I am current using `Model::encryptUsing()` to set a custom encrypter for eloquent so that I can use a  different encryption key for eloquent data. 

The challenge is that it's only used for  'encrypted', 'encrypted:array', 'encrypted:collection', 'encrypted:json', 'encrypted:object' casts and not the AsEncryptedArrayObject, AsEncryptedCollection castables 

I looked up the code and it seems like "expected behaviour" since the "encrypted:*" cast version first checks for the existence of a custom encrypter while the Castable classes don't
 
<img width="864" alt="Screenshot 2023-05-26 at 00 58 40" src="https://github.com/laravel/framework/assets/33207024/98b24b15-6b48-4223-98a1-a9999c05bec1">

<img width="1434" alt="Screenshot 2023-05-26 at 01 18 58" src="https://github.com/laravel/framework/assets/33207024/08b5ef2b-3159-4067-8587-082406f22a8a">

I am curious why the same logic is not used for both cases since;

1) It defeats the whole purpose of using a different encryption key for eloquent models since some casts will still use the global app key. 

2) Makes your code less resilient since refactoring from a cast say "encrypted:array" to AsEncryptedArrayObject would simply break if one is using a custom encrypter. 

<img width="1072" alt="Screenshot 2023-05-26 at 01 13 08" src="https://github.com/laravel/framework/assets/33207024/90c520dd-5b26-4d31-93be-56fcb9d1c502">

**Comment by Radiergummi** [comment]
This seems like a bug rather than an idea - maybe open an issue instead?
